src/EnergyPlus/PlantLoopHeatPumpEIR.cc

```diff
@@ -1292,4 +1299,1372 @@ void EIRPlantLoopHeatPump::oneTimeInit(EnergyPlusData &state)
+                    if (thisPLHP.auxElecEIRFoTempCurveIndex == 0) {
+                        ShowSevereError(state,
+                                        format("Invalid curve name for EIR FFHP (name={}; entered curve name: {}",
+                                               thisPLHP.name,
+                                               auxEIRFTName.get<std::string>()));
+                        errorsFound = true;
+                    }
+                } else {
+                    thisPLHP.auxElecEIRFoTempCurveIndex = 0;
+                }
+
+                // A18 auxiliary_electric_energy_input_ratio_function_of_plr_curve_name
+                if (fields.find("auxiliary_electric_energy_input_ratio_function_of_plr_curve_name") != fields.end()) {
+                    auto &auxEIRFPLRName = fields.at("auxiliary_electric_energy_input_ratio_function_of_plr_curve_name");
+                    thisPLHP.auxElecEIRFoPLRCurveIndex =
+                        Curve::GetCurveIndex(state, UtilityRoutines::MakeUPPERCase(auxEIRFPLRName.get<std::string>()));
+                    if (thisPLHP.auxElecEIRFoPLRCurveIndex == 0) {
+                        ShowSevereError(state,
+                                        format("Invalid curve name for EIR FFHP (name={}; entered curve name: {}",
+                                               thisPLHP.name,
+                                               auxEIRFPLRName.get<std::string>()));
+                        errorsFound = true;
+                    }
+                } else {
+                    thisPLHP.auxElecEIRFoPLRCurveIndex = 0;
+                }
+
+                // N13 standby_electric_power
+                if (fields.find("standby_electric_power") != fields.end()) {
+                    thisPLHP.standbyElecPower = fields.at("standby_electric_power").get<Real64>();
+                } else {
+                    Real64 defaultVal = 0.0;
+                    if (!state.dataInputProcessing->inputProcessor->getDefaultValue(
+                            state, cCurrentModuleObject, "standby_electric_power", defaultVal)) {
+                        // this error condition would mean that someone broke the input dictionary, not their
+                        // input file.  I can't really unit test it so I'll leave it here as a severe error
+                        // but excluding it from coverage
+                        ShowSevereError(state,                                                                            // LCOV_EXCL_LINE
+                                        "EIR FFHP: standby electric power not entered and could not get default value."); // LCOV_EXCL_LINE
+                        errorsFound = true;                                                                               // LCOV_EXCL_LINE
+                    } else {
+                        thisPLHP.standbyElecPower = defaultVal;
+                    }
+                }
```

Comment:
I do like seeing examples of how to find a default for a field past min-fields, but both of these fields are within min-fields and have a default. How is it possible that the field for Standby Electric Power is 1) not present, and 2) doesn't have a value? I don't think you need the `else` at line 2348. This may also mean that any fields that have a default don't even need to use `if (fields.find("standby_electric_power") != fields.end())`, just grab the value using `thisPLHP.standbyElecPower = fields.at("standby_electric_power").get<Real64>();`.

    HeatPump:AirToWater:FuelFired:Heating,
      \min-fields 31
      A18, \field Auxiliary Electric Energy Input Ratio Function of PLR Curve Name
      N13; \field Standby Electric Power
           \default 0

    HeatPump:AirToWater:FuelFired:Cooling,
      \min-fields 26
      A16, \field Auxiliary Electric Energy Input Ratio Function of PLR Curve Name
      N10; \field Standby Electric Power
           \default 0

Reply:
Just one question: if say hypothetically , the min-fields were not set in this way as needing full number of inputs, would this simplified way of `thisPLHP.standbyElecPower = fields.at("standby_electric_power").get<Real64>();` still be valid? In this "hypothetical" case, the major concern would if this field is left blank, then would some extra code needed to deal with the situation?

If extra code is need to deal with the "blank field" situation on this field, probably it would be more robust to assume that the min-fields are not full---e.g. in the future some changes may be made to allow fewer number of min-fields?

Reply:
Well, that's true but then min-fields would have little meaning. I have begun removing code that is covered by the rule set in the IDD.

Reply:
1. Defaults are currently only applied when using `GetObjectItem` up to \min-fields!
2. Raw epJSON does not currently set any defaults and does not care about \min-fields!!
3. \min-fields is used by Transition, IDFEditor, `GetObjectItem` and possibly other tools.
4. Some developers simply set the default on the struct (which requires making sure it's consistent with the IDD default) and skip the field if it's not there in epJSON.
5. Some developers use `getDefaultValue`.
6. Some developers use `getAlphaFieldValue`, `getRealFieldValue` etc. with its quirky setup requirements. These helper functions check for blank and return the default (regardless of \min-field).

Reply:
@mjwitte this is a great summary of the status of defaults...  

Great in a similar way that ~~Voldemort~~ did great things.  Terrible...but great.

Reply:
🤦